### PR TITLE
Memory Optimizations and Fix for Resource Leaks

### PR DESCRIPTION
**Pull request by samirsahu1947**
Fix for Resource Buffer Leak in libtuv

### CODE REVIEW

**Comment by seinfra** [comment]
Code Rule Check OK.

File: external/libtuv/source/unix/uv_unix_getaddrinfo.c [hunk 1]
```diff
@@ -220,10 +220,12 @@ int uv_getaddrinfo(uv_loop_t *loop, uv_getaddrinfo_t *req, uv_getaddrinfo_cb cb,
 
 	if (cb) {
 		uv__work_submit(loop, &req->work_req, uv__getaddrinfo_work, uv__getaddrinfo_done);
+		free(buf);
 		return 0;
 	} else {
 		uv__getaddrinfo_work(&req->work_req);
 		uv__getaddrinfo_done(&req->work_req, 0);
+		free(buf);
```
Comment (sunghan-chang) [inline external/libtuv/source/unix/uv_unix_getaddrinfo.c]: Inside uv__getaddrinfo_done , there are free codes for buf. Is it duplicated?

**Comment by sunghan-chang** [comment]
@samirsahu1947 Please add module information at commit title.

**Comment by sunghan-chang** [comment]
@samirsahu1947 As we discussed, this is wrong. What is your plan?

File: external/libtuv/source/unix/uv_unix_getaddrinfo.c [hunk 2]
```diff
@@ -220,10 +220,12 @@ int uv_getaddrinfo(uv_loop_t *loop, uv_getaddrinfo_t *req, uv_getaddrinfo_cb cb,
```
Comment (jsdosa) [inline external/libtuv/source/unix/uv_unix_getaddrinfo.c]: i think, you should move "free(buf)" here.

**Pull request closed without merging**